### PR TITLE
Adding a helpful script to automate some of the tedium around doing updates

### PR DESCRIPTION
**Pull request by blast-hardcheese**
See usage section at the top of the script. Once I get a better feel around usability I'll make this more streamlined, but for now this should be better than before.

### CODE REVIEW

**Comment by blast-hardcheese** [comment]
Failing check here addressed in https://github.com/replit/pyright-extended/pull/34

File: scripts/set-metadata.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        contributors_json="$2"; shift 2 || die 'Missing contributors'
+        ;;
+        --publisher)
+        publisher="$2"; shift 2 || die 'Missing publisher'
+        ;;
+        *)
+        die 'Unknown arguments: $@'
+        break
+        ;;
+    esac
```
Comment (cdmistman) [inline scripts/set-metadata.sh:45]: nit: please indent

File: scripts/set-metadata.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            --argjson contributors "$contributors_json" \
+            --arg publisher "$publisher" \
+            '.
+            | .name |= $name
+            | .displayName |= $displayName
+            | .description |= $description
+            | .version |= $version
+            | .author |= (.name |= $author)
+            | .contributors |= (if $contributors == null then empty else $contributors end)
+            | .publisher |= $publisher
```
Comment (cdmistman) [inline scripts/set-metadata.sh:64]: neat!

File: scripts/set-metadata.sh [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    name="$1"; shift || die 'Missing name'
+    version="$1"; shift || die 'Missing version'
+
+    contents="$(
+        jq --indent 4 \
+            --arg name "$name" \
+            --arg version "$version" \
+            '.name |= $name
+            | .version |= $version
+            | .packages[""] |= (.name |= $name | .version |= $version)' \
```
Comment (cdmistman) [inline scripts/set-metadata.sh:82]: is this necessary? can't we implicitly do this eg with `npm i`

**Review by cdmistman — APPROVED** [review]

**Pull request merged**